test/integration/tcp_tunneling_integration_test.cc

```diff
@@ -67,14 +67,26 @@ class ConnectTerminationIntegrationTest : public HttpProtocolIntegrationTest {
     EXPECT_EQ(downstream_received_data, response_->body());
   }
 
-  Http::TestRequestHeaderMapImpl connect_headers_{{":method", "CONNECT"},
-                                                  {":path", "/"},
-                                                  {":protocol", "bytestream"},
-                                                  {":scheme", "https"},
-                                                  {":authority", "foo.lyft.com:80"}};
+  Http::TestRequestHeaderMapImpl connect_headers_h1_{{":method", "CONNECT"},
+                                                     {":authority", "foo.lyft.com:80"}};
+
+  // The client H/2 codec expects the request header map to be in the form of H/1
+  // upgrade to issue an extended CONNECT request
+  Http::TestRequestHeaderMapImpl extended_connect_headers_h2_{
+      {":method", "GET"},        {":path", "/"},       {"upgrade", "bytestream"},
+      {"connection", "upgrade"}, {":scheme", "https"}, {":authority", "foo.lyft.com:80"}};
+
   void clearExtendedConnectHeaders() {
-    connect_headers_.removeProtocol();
-    connect_headers_.removePath();
+    extended_connect_headers_h2_.setMethod("CONNECT");
+    extended_connect_headers_h2_.removeConnection();
+    extended_connect_headers_h2_.removePath();
+    extended_connect_headers_h2_.removeUpgrade();
+  }
+
+  Http::RequestHeaderMap& getRequestHeaderMap() {
```

Comment:
Why not test regular connect and extended connect explicitly in tests, and early-return for explicit extended connect for HTTP/1? I think it would be more clear to have the e2e test verify standard connect for most tests, and explicitly set use_extended_connect_ for tests where we want to test that.

Reply:
Makes sense. I think all of the tests can use standard CONNECT (and they did so in the past anyway with the client codec sanitization). I have added a test to validate that Envoy interprets extended CONNECT with bytestream protocol as a standard CONNECT request.